client/src/components/App.jsx

```diff
@@ -1,58 +1,93 @@
 import React from 'react';
+import axios from 'axios';
 import Reviews from './Reviews/Reviews.jsx';
 import ProductOverview from './ProductDetails/ProductOverview.jsx';
 import QuestionList from './QandA/QuestionList.jsx';
 import RelatedItems from './RelatedItems/RelatedItems.jsx';
 import NavBar from './Navbar.jsx';
 
-import Reviewsdata from './Reviews/reviewsData.js';
-import QuestionListdata from './QandA/QandAListData.js';
-import relatedItemsdata from './RelatedItems/relatedItemsData.js';
-import productListDummyData from './ProductDetails/sampleData/productListDummyData.js';
-
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      currentProductID: '39333',
+      currentProductID: '',
       currentStyleID: 234004,
-      productStylesArray: productListDummyData.productStyles.results,
-      currentProduct: productListDummyData.productToDisplay,
+      productStylesArray: [],
+      currentProduct: [],
       reviews: [],
-      questionList: QuestionListdata,
-      relatedItems: relatedItemsdata,
-
+      questionList: [],
+      relatedItems: [],
     };
+    this.cachedProducts = {};
     this.changeCurrentStyle = this.changeCurrentStyle.bind(this);
+  }
 
+  componentDidMount() {
+    this.getProductData('39333');
   }
 
+  getProductData(productId) {
+    let cachedProduct = this.cachedProducts[productId];
+    if (cachedProduct) {
+      let {currentProduct, productStylesArray, reviews, questionList} = cachedProduct;
+      this.setState({
+        currentProductID: productId,
+        currentProduct,
+        productStylesArray,
+        reviews,
+        questionList
+      });
+    } else {
+      axios.get(`./products/${productId}`)
+        .then(({data}) => {
+          let productInfo = {
+            currentProduct: data[0],
+            productStylesArray: data[1].results,
+            reviews: {
+              reviewsMetadata: data[4],
+              reviews: data[3],
+            },
+            questionList: data[2]
+          };
+          this.cachedProducts[productId] = productInfo;
+          this.getProductData(productId);
+        })
+        .catch(err => {
+          alert(`Error ${err} occurred`);
```

Comment:
Glad you have an error handler! -> this is me being REALLY nitpicky, but I would format it along the lines of previous work we've done (`Error encountered: ${err}`) as the words 'Error' and 'occurred' will be split by a large body of text/JSON data and no longer be grammatical.

Reply:
If I'm not mistaken, err will be the http status code sent by the server. So it'll render 'Error 503 occurred' in this case. 
But I can make the suggested change.